legion/engine/core/scenemanagement/scenemanager.cpp

```diff
@@ -0,0 +1,25 @@
+#include <core/scenemanagement/scenemanager.hpp>
+#include <core/scenemanagement/scene.hpp>
+namespace legion::core::scenemanagement
+{
+    std::unordered_map < id_type, std::string> SceneManager::sceneNames;
+    std::unordered_map<id_type, ecs::component_handle<scene>> SceneManager::sceneList;
+    ecs::EcsRegistry* SceneManager::registry;
+
+
+
+    void SceneManager::createScene(const std::string& name)
+    {
+        scene s;
+        s.id = nameHash(name);
+        sceneNames.emplace(s.id, name);
+        auto sceneHandle = registry->createEntity().add_component(s);
+        sceneList.emplace(nameHash(name), sceneHandle);
+    }
```

Comment:
maybe add an overload that allows you to pass an already existing entity

Reply:
like a copy method?